src/nb_enum.cpp

```diff
@@ -107,6 +117,48 @@ static PyObject *nb_enum_init(PyObject *, PyObject *, PyObject *) {
     return 0;
 }
 
+static bool nb_enum_set_value(PyObject *self, PyObject *value) {
+    nb_inst *inst = reinterpret_cast<nb_inst*>(self);
+    if (inst->ready) {
+        PyErr_SetString(PyExc_ValueError, "can't change enum value");
+        return false;
+    }
+    enum_supplement &supp = nb_enum_supplement(Py_TYPE(self));
+    size_t size = nb_type_data(Py_TYPE(self))->size;
+    void *inst_data = inst_ptr(inst);
+    const char *too = nullptr;
+    if (supp.is_signed) {
+        long long lval = PyLong_AsLongLong(value);
```

Comment:
I also tried switching on `size` and calling the appropriate `load_[iu](8|16|32|64)` function, but that was larger in code size.